docs/source/jit_language_reference.rst

```diff
@@ -346,6 +348,74 @@ like any other TorchScript type:
     print(sum_pair(p))
 
 
+.. _TorchScript Enum:
+.. _TorchScript Enums:
+.. _torchscript-enums:
+
+TorchScript Enums
+^^^^^^^^^^^^^^^^^^^
+
+.. warning::
+
+    TorchScript enum support is experimental..
+
+Python enums can be used in TorchScript without any extra annotation or code:
+
+::
+
+    from enum import Enum
+
+
+    class Color(Enum):
+        RED = 1
+        GREEN = 2
+
+    @torch.jit.script
+    def enum_comp(x: Color, y: Color) -> bool:
+        return x == y
+
+After an enum is defined, it can be used in both TorchScript and Python interchangeably
+like any other TorchScript type:
+
+::
+
+    from enum import Enum
+
+    class Color(Enum):
+        RED = 1
+        GREEN = 2
+
+    @torch.jit.script
+    def return_enum(cond: bool):
+        if cond:
+            return Color.RED
+        else:
+            return Color.GREEN
```

Comment:
And then maybe include this example where you access the field name `Color.RED` or make it part of the other function